### ISSUE TITLE
Background location permission on Android 10

### ISSUE THREAD

**Issue by leso-kn** [nift4/microg_installer_revived#9]
Android 10 introduced a new 'background location' permission.

As mentioned [here](https://old.reddit.com/r/MicroG/comments/f66d9c/network_location_unifiednlp_in_android_10_q/) UnifiedNlp needs to request this permission in order to function propperly.

Being a very simple fix i would love to see it in Microg Installer Revived :)

---

To fix it add

```xml
<uses-permission android:name="android.permission.ACCESS_BACKGROUND_LOCATION"/>
```

to the AndroidManifest.xml of GmsCore.

There's also a open [pull request](https://github.com/microg/android_packages_apps_GmsCore/pull/1004) on microg that does exactly that.

**Comment by nift4** [comment]
Hi. Thanks for using my module :)
Sadly I don't understand what do you want. microG Installer Revived uses the offical prebuilt APKs and I can't simply change the AndroidManifest of that apk. Is your suggestion to include an APK of an fork with it fixed?

**Comment by leso-kn** [comment]
Hi! :)
I see. Sorry, I was not sure if the official or a custom apk is bundled with the package.

I've read that the developer of microG recently stated he would pick up the work on it soon. But that's already some time ago again.

If including an apk of a fork is an option I can check if I find one that works. Since there's no updates to the official apk right now it might be worth including a fixed version for the time being. What do you think?

**Comment by nift4** [comment]
I agree. When you find an fork with it fixed and an apk I change it. I would be happier if you change it via a pull request through, you only need to swapout apk. I already added the permission in the other (module) xml as it's needed anyway

**Comment by nift4** [comment]
Nice, I wait for the apk or pull request :) (In meantime I work on Play Store)

**Comment by nift4** [comment]
Hi again; would it be for you possible to test something?

**Comment by leso-kn** [comment]
Well, I decided to use the fork of @eplinux as it already contains the fixes for both GmsCore and UnifiedNlp.
Right now I am fighting with serval docker setups in order to compile the repo; the gradle files seem to be a bit outdated.

Once it's running it should be no problem to test it :)

**Comment by leso-kn** [comment]
I already thought about just de- and recompiling the official apk with `apktool`.
But this would be a rather dirty solution. I might re-consider it for testing in case fixing the gradle stuff takes very long.

**Comment by leso-kn** [comment]
Ah wait! Nevermind. I thought i need to compile something for #10.
Yes, i can update the module and tell you if it worked.

Btw, feel free to contact me on Telegram if you want. I guess that might make communication a bit easier (I'm from germany too) https://t.me/leso_kn

**Comment by nift4** [comment]
From module site this is fixed and UnifiedNlp 2.0 is worked on, so ig I can close this
